src/cmr/concepts/draft.js

```diff
@@ -13,16 +15,58 @@ export default class Draft extends Concept {
   constructor(conceptType, headers, requestInfo, params) {
     super(conceptType, headers, requestInfo, params)
 
-    const singularConceptType = conceptType.replace('drafts', 'draft')
     const {
-      providerId,
-      ummVersion
+      draftConceptId = '',
+      providerId
     } = params
 
     this.ingestPath = `providers/${providerId}/${conceptType}`
+    this.publishPath = `publish/${draftConceptId}`
+
+    const {
+      ummCollectionVersion,
+      ummServiceVersion,
+      ummToolVersion,
+      ummVariableVersion
+    } = process.env
+
+    let ummType
+    let ummName
+    let ummVersion
+    switch (conceptType) {
```

Comment:
Can't the sub classes just set/return these values instead of having a switch statement in the parent?

Reply:
There are no sub classes